src/cargo/ops/cargo_compile.rs

```diff
@@ -191,6 +195,19 @@ pub fn compile_pkg<'a>(package: &Package,
     let targets = target_with_args.as_ref().map(|&(t, ref p)| vec![(t, p)])
                                            .unwrap_or(targets);
 
+    let mut target_with_rustdoc = None;
+    if let Some(args) = *extra_rustdoc_args {
+        let mut target_with_rustdoc_inner = Vec::new();
+        for &(target, profile) in &targets {
```

Comment:
I think you'll probably want to filter this by `profile.doc` as not all targets have rustdoc run on them by default.